Adds API for direct binary access

mattvryan opened the pull request:
This supports an optional feature to do direct binary upload and download as described in https://wiki.apache.org/jackrabbit/Direct%20Binary%20Access.  This change includes new interfaces and exceptions that comprise the API for this feature.

reschke committed:
JCR-4090: fix javadoc problems - fix "exception not thrown"

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776421 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare branch 2.14

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776533 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776535 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4067: remove HttpClient3 based WebDAV API

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776576 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4094: Fix Javadoc for JDK 1.8 doctool compliance - remove TCK related custom tags - @test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776907 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4094: Fix Javadoc for JDK 1.8 doctool compliance - remove TCK related custom tags - @executeClass

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776911 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4094: Fix Javadoc for JDK 1.8 doctool compliance - remove TCK related custom tags - @keywords

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776912 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4094: Fix Javadoc for JDK 1.8 doctool compliance - remove TCK related custom tags - @sources

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776914 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4094: Fix Javadoc for JDK 1.8 doctool compliance - remove TCK related custom tags - @tck.config

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1776918 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4067: remove HttpClient3 based WebDAV API - fix Javadoc

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1777156 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4095: release Jackrabbit 2.15.0 - release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1777759 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1777949 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1777951 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4096: stop using deprecated GenericRepositoryFactory

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1778270 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4100: jackrabbit-webdav: Scope implements equals() but not hashCode()

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1779166 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4102: jcr2dav: change polling thread sometimes dies with NPE

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1779460 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4104: cleanup how command lines for test executions are constructed

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1779614 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4101: add code coverage profile (jacoco)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1779632 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4106: fix JCR API docs link

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1780208 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4065: Add WebDAV API based on HttpClient4 - add API docs linkage

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1780220 13f79535-47bb-0310-9956-ffa450edef68

Commit:
JCR-4093 IndexRule are meant to be applied based on both primaryType and minin type based inheritance. Currently it appears that only primaryType based inheritance is working

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1780335 13f79535-47bb-0310-9956-ffa450edef68

Commit:
JCR-4093 IndexRule are meant to be applied based on both primaryType and minin type based inheritance. Currently it appears that only primaryType based inheritance is working

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1780336 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1780384 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4107: Upgrade to httpclient 4.5.3

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1780406 13f79535-47bb-0310-9956-ffa450edef68

Commit:
JCR-4110 The JackrabbitQueryParser uses the wrong Lucene-Version in its initializing

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1782145 13f79535-47bb-0310-9956-ffa450edef68

Commit:
JCR-4110 The JackrabbitQueryParser uses the wrong Lucene-Version in its initializing

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1782173 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
JCR-4115: Don't use SHA-1 for new DataStore binaries (Jackrabbit)

- Using SHA-256 instead of SHA-1 to create blob record ids


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1785225 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4117: Release Jackrabbit 2.15.1

Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1785264 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.1

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1785460 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1785462 13f79535-47bb-0310-9956-ffa450edef68

mebigfatguy committed:
once the property is found, break out of the loop

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1785579 13f79535-47bb-0310-9956-ffa450edef68

mebigfatguy committed:
use short-circuiting conditionals (&& vs &)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1785580 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4118: RepositoryChecker creates invalid node names

create valid names / do sanity check in test case

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1786325 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4119: Upgrade httpcomponents/httpmime to 4.5.3

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1786330 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
JCR-4121: ConcurrentModificationException in InternalVersionHistoryImpl.fixLegacy()

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1787043 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4122: align parent pom references with Oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1787381 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4127: update to latest apache parent pom (18)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1792066 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4128: update maven plugins and require Maven 3.2.1

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1792100 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4128: update maven plugins and require Maven 3.2.1

maven-scr-plugin

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1792105 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4128: update maven plugins and require Maven 3.2.1

maven-assembly-plugin

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1792113 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4129: get rid of unused org.json dependency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1792193 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4112: Require Java 8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1792695 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4132: Release Jackrabbit 2.15.2


Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1792702 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4133: fix javadoc problems that are errors with JDK8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793315 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4132: Release Jackrabbit 2.15.2


Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793317 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4133: fix javadoc problems that are errors with JDK8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793323 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4133: fix javadoc problems that are errors with JDK8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793327 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4133: fix javadoc problems that are errors with JDK8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793332 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4133: fix javadoc problems that are errors with JDK8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793339 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793344 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793346 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4134: update build instructions wrt Java versions

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793458 13f79535-47bb-0310-9956-ffa450edef68

mebigfatguy committed:
guard against NPEs

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1793639 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-3901: TCK LockManagerTest does not allow new JCR 2.0 functionality for lock token transfers

Introduce nex TCK config property, defaulting to the previous behavior, allowing to configure tests to check that open scoped lock tokens can be shared.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1796980 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4139: Update commons-fileupload dependency to 1.3.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1797209 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4140: Update easymock dependency to 3.4

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1797215 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4142: update junit dependency to 4.12

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1797917 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4143: update findbugs dependency to 3.0.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1797921 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4137: Release Jackrabbit 2.15.3 - Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1798059 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.3

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1798192 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1798194 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4145: upgrade clirr plugin to 2.8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1798586 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4147: clirr profile needs to use sensible baselines - use 2.14.0 as baseline version

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1798666 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4148: revert update of findbugs

(undoes change for JCR-4143)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1798862 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4146: json extension is removed by AbstractWebdavServlet on COPY request

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1799429 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
JCR-4149: change to drop SHA-1 requires version change

- Add back the protected property in DbDataStore
- Update version of org.apache.jackrabbit.core.data.db - 2.13.6
- Update version of org.apache.jackrabbit.core.data - 2.14.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1799538 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4150: enable bundle:baseline check

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1799549 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
JCR-4144 : JackrabbitAccessControlList should have an API boolean isMultiValueRestriction(restrictionName)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1799557 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4151: remove clirr profile in branches where bundle:baseline is used

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1799575 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4152: Release Jackrabbit 2.15.4

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1799577 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.4

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1799658 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1799660 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4154: davex upload of binaries broken

add test case/fix multipart format

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1800359 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4154: davex upload of binaries broken

refactored test case/changed filename param to something meaningful

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1800378 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4157: jackrabbit-jcr-server: code cleanup

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1800752 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4164: add test coverage for content codings

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1802925 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4158: jackrabbit-server doesn't handle content-codings properly

Reject all requests with non-empty Content-Encoding header fields with status 415.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1802977 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4170: release Jackrabbit 2.15.5 - candidate release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1803148 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.5

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1803430 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1803432 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4177: move Mockito version to parent pom

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1807234 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4173: Unable to receive observation events when connecting via DavEx

make sure we do not produce broken XML, add a minimal test case

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1807244 13f79535-47bb-0310-9956-ffa450edef68

rombert committed:
JCR-4179 - Test failures with Java 9 when using javax.imageio.spi.ServiceRegistry

Switched from javax.imageio.spi.ServiceRegistry to java.util.ServiceLoader

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1807321 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4180: Release Jackrabbit 2.15.6 - Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1807344 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.6

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1807563 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1807565 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4185: spi2dav URIResolverImpl should check status code before attempting to parse multistatus

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1808752 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4184: migrate from org.mortbay.jetty to org.eclipse.jetty

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1808754 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4186: Use current Derby version

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1809149 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4188: avoid use of sun.security.acl.GroupImpl in PrincipalManagerTest

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1809329 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4190: maven-assembly-plugin:2.6:single failing with Java 9 - consistently use 3.0.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1809624 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-3929: ConsistencyChecker now deals with empty repositories

(thanks to pacuk.anton@gmail.com for investigating)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1810108 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4182: new release checksum requirements - changed file extension from SHA to SHA1

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1811667 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4078: occasional test failure in GarbageCollectorTest.testSimultaneousRunGC

Improve synchronization.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812515 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4196: update surefire and failsafe plugins for use with java 9

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812543 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4195: jackrabbit-jcr-rmi - stop using RMIC, update export version numbers

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812554 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4200: javax.transaction.UserTransaction hidden by surefire plugin in with Java 9

Override surefire's default "--add-module" directive for Java 9 (to be revisited in future versions of Java and/or the surefire plugin)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812634 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4201: Release Jackrabbit 2.15.7 - candidate release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812717 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4201: Release Jackrabbit 2.15.7 - candidate release notes - change backed out

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812721 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4201: Release Jackrabbit 2.15.7 - candidate release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812722 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.7

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812878 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812880 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4205: Upgrade JTA dependency to javax.transaction-api 1.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1812994 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4206: Switch to official OSGi versioning annotations

Part 1: just version annotations

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1813029 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4206: Switch to official OSGi versioning annotations

Part 2: version annotations in jackrabbit-api

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1813064 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4206: Switch to official OSGi versioning annotations

Part 3: other annotations in jackrabbit-api

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1813098 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4191: baseline checks fails for jackrabbit-webdav under Java 9

fixed by bumping the minor version number

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1813148 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4207: Use Derby 10.14.1.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1813150 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4211: Release Jackrabbit 2.15.8 - candidate release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1814728 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4212: add SHA512 checksums to releases

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1814831 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4211: Release Jackrabbit 2.15.8 - candidate release notes updated

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1814834 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.15.8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1815015 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1815017 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1815643 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare branch 2.16

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1815780 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1815782 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4218: switch bundle comparisonVersion

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1816227 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4217: Release Jackrabbit 2.17.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1816376 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4221: Upgrade Apache HttpComponents to 4.5.4

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817094 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4223: Upgrade commons-fileupload dependency to 1.3.3

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817097 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4224: Upgrade tomcat-servlet dependency to 7.0.82

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817098 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4225: Upgrade commons-chains dependency to 1.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817100 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4226: Upgrade tika-parsers dependency to 2.16

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817113 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4228: Update Oak dependency to latest 1.0 stable release

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817201 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4231: Upgrade aws-java-sdk-s3 dependency to 1.11.241

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817213 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4222: Document reduced RMI interop with older servers after java-9 related changes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817341 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4078: unreliable test excluded

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817373 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4233: Update H2DB test dependency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1817377 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4217: Release Jackrabbit 2.17.0 - Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1818142 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.17.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1818506 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1818508 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4236: remove obsolete textFilterClasses config parameters

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1818586 13f79535-47bb-0310-9956-ffa450edef68

mebigfatguy committed:
JCR-4237 - cleanup throws clauses of private methods

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1819269 13f79535-47bb-0310-9956-ffa450edef68

mebigfatguy committed:
JCR-4238: use StandardCharsets to avoid having to handle UnsupportedEncodingException

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1819271 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4239: Suppress Tika startup warnings

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1819839 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4240: IndexingQueueTest relies on Tika behavior that is changed in Tika 1.17

Changed test to use an actual payload (and parse that), adding tika config to detect the custom type. (Thanks to tallison@apache.org for the proposed fix!)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1819849 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
JCR-4241: Jacoco plugin fails with exception

Do not prepare agent when coverage is skipped

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820119 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
JCR-4242: Build failure: unapproved license



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820133 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4239: Suppress Tika startup warnings

(fix default for 'dynamic')

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820294 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4244: Upgrade tomcat dependency to 8.5.24

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820573 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820647 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820649 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820653 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820654 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820656 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log on LinkageError

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820657 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4215: Use Tika version 1.17

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820675 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log (No Tabs)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820737 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820744 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820746 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820747 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820748 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4229: Remove Oak dependency in webapp

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1820753 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4247: Update commons-io dependency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1821247 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4248: Upgrade httpcore dependency to 4.4.9

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1821475 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4001: When using Node.getProperties(String namePattern) also child nodes are processed

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1821597 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4250: remove guava dependency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1821705 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4252: Upgrade httpclient dependency to 4.5.5

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1821880 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
JCR-4249 Introduce replacement for java.security.acl.Group



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822494 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4251: Release Jackrabbit 2.17.1 - Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822514 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4251: Release Jackrabbit 2.17.1 - Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822516 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.17.1

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822540 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822542 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4254: Update Logback version to >= 1.2.0, SLF4J accordingly

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822643 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4253: RepositoryConfig: add some handling for mkdir failure

(patch by diego.schivo - thanks)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822863 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4256: create announcement mail template for releases

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822947 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4256: create announcement mail template for releases (fix line ends, add a note that the template was generated)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1822950 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
JCR-4258 Remove SimpleJBossAccessManager in preparation of upcoming acl changes



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1824103 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4261: webapp: align jsons-simple dependencies internally and with oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1824756 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4262: jcr-server: align org.apache.felix.scr.annotations with oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1824763 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4263: jcr-server, jackrabbit-bundle: align org.osgi dependencies with oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1824764 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4264: jackrabbit-standalone: align commons-cli dependency with oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1824771 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4215: Use Tika version 1.17

(remove duplicate dependency)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1824876 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4266: Update servlet-api to 2.5

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1824897 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4271: Remove redundant JavaDoc link to Java 7 API

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1826154 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4272: Upgrade surefire and failsafe plugins to 2.21.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1826178 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4273: Mention SHA512 checksums in release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1826230 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4274: stop advertising MD5 checksums

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1826647 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4275: fix incorrect link to KEYS in release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1826940 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4277: stop creating MD5 checksums for releases

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1826964 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
JCR-4246 java.security.acl deprecated in Java 10, marked for removal in Java 11



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1827242 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
JCR-4246 java.security.acl deprecated in Java 10, marked for removal in Java 11
 - license header



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1827263 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4280: code coverage checks fail on Java 10

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1828213 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4282: release Jackrabbit 2.17.2 - candidate release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1828215 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.17.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1828339 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1828341 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4285: fix Javadoc warnings - empty paragraphs

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1828804 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4285: fix Javadoc warnings - no param

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1828815 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1829246 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Debug Log

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1829247 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4288: Upgrade tika-parsers dependency to 2.18

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1830107 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4290: remove unused commons-codec dependency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1830201 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4291: FileInputStream for workspace.xml not closed in RepositoryConfig.loadWorkspaceConfig(File)

(patch by Jan Quadflieg)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1830540 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4292: davex: preserve cause in exceptions and log affected URI

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1830753 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4293: jackrabbit-core: observation tests should not rely on mix:lockable mixin type

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1830814 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4294: TCK tests should pass on repositories without locking support

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1830878 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4296: Upgrade httpmime dependency to 4.5.5

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1830951 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4299: Update derby dependency to 10.14.2.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1831182 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4297: Release Jackrabbit 2.17.3 - Candidate Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1831187 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.17.3

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1831361 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1831363 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4302: BTreeManager: fix Eclipse compiler error

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1831854 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4304: update Jetty to supported version 9.2.*

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1831860 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4307: Update animal-sniffer-maven-plugin to 1.16

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1832058 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4306: switch to findbugs replacement that is still maintained (spotbugs)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1832090 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4301: get rid of JSR 305 dependency - switch to JetBrains annotations

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1832177 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4312: set baseline comparisonVersion to latest stable (2.16.1)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1832690 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4256: create announcement mail template for releases (adding missng line breaks)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1833374 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4316: set baseline comparisonVersion to latest stable (2.16.2)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1833582 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4317: davex remoting fails for non-ASCII characters in node names

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1833835 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4318: Update failsafe and surefire plugin versions to 2.22.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1833891 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-3211: Support HTTP proxy in SPI2DAV

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1834008 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4320: Update spotbugs plugin to 3.1.5

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1834308 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4321: Update maven plugins from org.apache.maven.plugins

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1834315 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4322: Consistent use of log4j versions

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1834401 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4323: webapp: update Tomcate dependency to 8.5.32

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1834418 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4324: NPE on Version.getLinearPredecessor() implementation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1834424 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Adds HttpBinaryProvider interface and supporting cast.

This supports an optional feature to do direct binary upload
and download.

reschke commented on jackrabbit-api/pom.xml:
```diff
@@ -70,5 +70,9 @@
     </dependency>
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
```
We switched to idea annotations recently (see https://issues.apache.org/jira/browse/JCR-4301).

reschke committed:
JCR-4326: Update aws java sdk version to 1.11.330 (consistent with Oak)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1834673 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4327: Update httpcore dependency to 4.4.10

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1834922 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4329: Release Jackrabbit 2.17.4 - candidate release notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1835211 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4331: Update httpclient dependency to 4.5.6

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1835445 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4332: Update httpmime dependency to 4.5.6

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1835448 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
JCR-4333: Update javax.transaction dependency to 1.3

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1835459 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Use IDEA annotations instead of Google FindBugs

mattvryan commented:
@reschke I have made the change to use the idea annotations; any better?

reschke committed:
JCR-4329: Release Jackrabbit 2.17.4 - Release Notes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1835508 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare release jackrabbit-2.17.4

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1835512 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1835514 13f79535-47bb-0310-9956-ffa450edef68

Commit:
API updates for direct binary access based on ValueFactory approach.

Commit:
Move completeBinaryUpload method to JackrabbitValueFactory

Commit:
Rename to match direct access naming

Commit:
Changes URL usage to URI, adds Javadoc

Commit:
Revert accidental change from refactor

Commit:
Add back AccessDeniedException

Commit:
Adds API for specifying response headers in download

Commit:
Code cleanup per review

Commit:
Merge branch 'direct-binary-access' into direct-binary-access-with-OAK-7637

Commit:
Fix BinaryDownloadOptions to support all required header capabilities

Commit:
Minor changes to BinaryDownloadOptions primarily.

- contentType=>mimeType
- contentTypeEncoding=>encoding
- Adds javadoc for BinaryDownloadOptions and
  BinaryDownloadOptionsBuilder
- Makes BinaryDownloadOptions and BinaryDownloadOptionsBuilder
  final classes.

Commit:
Clarifying edits on JavaDocs.

Commit:
Clarifications to Javadocs for BinaryDownloadOptions

Commit:
Merge branch 'devel' into direct-binary-access

Commit:
Updated Javadoc with more usage detail and examples

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryUpload.java:
```diff
@@ -0,0 +1,211 @@
+ *     then the client cannot proceed and will have to request a new set of URIs
+ *     with the right fileSize as maxSize
+ *     <li>If fileSize is smaller than minPartSize, then take the first provided
+ *     upload URI to upload the entire binary, with partSize = fileSize</li>
```
shouldn't this be maxPartSize?

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryUpload.java:
```diff
@@ -0,0 +1,211 @@
+ *     next URI from uploadURIs (strictly in order), proceed with a standard
+ *     HTTP PUT for each, and for the last part use whatever segment size is
+ *     left
+ *     <li>If a segment fails during upload, retry (up to a certain time out)
```
HTTP PUT if a HTTP(s) URI, otherwise to-be-determined...

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/JackrabbitValueFactory.java:
```diff
@@ -0,0 +1,188 @@
+     * @return A {@link BinaryUpload} that can be used by the client to complete
+     *         the upload via a call to {@link #completeBinaryUpload(String)},
+     *         or {@code null} if the implementation does not support the direct
+     *         upload feature.
```
This should throw an exception instead of returning null.

(should we also define a repository descriptor maybe?)

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownload.java:
```diff
@@ -0,0 +1,64 @@
+     * <p>
+     * The implementation will attempt to apply the specified {@code
+     * downloadOptions} to the subsequent download.  For example, if the caller
+     * knows that the URL refers to a specific type of content, the caller can
```
URL -> URI

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownload.java:
```diff
@@ -0,0 +1,64 @@
+     *         if the caller wishes to accept the service provider's default
+     *         behavior.
+     * @return A URI for downloading the binary directly, or {@code null} if the
+     *         binary cannot be downloaded directly or if the underlying
```
Should throw instead of returning null.

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java:
```diff
@@ -0,0 +1,317 @@
+
+    /**
+     * Returns the MIME type that should be assumed for the binary that is to be
+     * downloaded.  This value should be a valid {@code jcr:mimeType}.  This
```
"MIME" -> "Internet Media Type"

FWIW, maybe we should also not copy JCR's splitting of type and charset, and just provide the complete value for the Content-Type header field.

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java:
```diff
@@ -0,0 +1,317 @@
+
+    /**
+     * Returns the encoding that should be assumed for the binary that is to be
+     * downloaded.  This value should be a valid {@code jcr:encoding}.  This
```
"encoding"  -> "character encoding"

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java:
```diff
@@ -0,0 +1,317 @@
+     * instance of this class.
+     *
+     * @return A String representation of the encoding, or {@code null} if no
+     *         encoding has been specified.
```
The encoding is a name, so I think that "String representation" is confusing here.

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java:
```diff
@@ -0,0 +1,317 @@
+     * BinaryDownloadOptionsBuilder#withDispositionTypeInline()} or {@link
+     * BinaryDownloadOptionsBuilder#withDispositionTypeAttachment()} when
+     * building an instance of this class.  The default value of this setting is
+     * "attachment".
```
Should be IMHO "inline".

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java:
```diff
@@ -0,0 +1,317 @@
+         * Note that if the MIME type is text-based, the caller may also wish to
+         * set the encoding which is done separately.  See {@link
+         * #withEncoding(String)}.
+         * <p>
```
"Note that if the internet media type defines a charset parameter, ...."

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java:
```diff
@@ -0,0 +1,317 @@
+         * Calling this method has the effect of instructing the service
+         * provider to set {@code encoding} as the encoding of the content type
+         * in the {@code Content-Type} header of the response to a request
+         * issued with a URI obtained by calling {@link
```
"header" -> "header field"

reschke commented on jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java:
```diff
@@ -0,0 +1,317 @@
+         * call to {@link #build()}.
+         * <p>
+         * If no filename is provided, no {@code Content-Disposition} header
+         * will be specified to the service provider.
```
I think we'll need to define the case of no filename but non-default disposition type.

reschke committed:
editorial fixes

reschke committed:
- fix javadoc build errors
- terminology (mime type -> internet media type, encoding -> character encoding)
- add links to RFC 6266
- make disposition type "inline" default
- break the connection between disposition type and filename

reschke committed:
add package level Javadoc

reschke committed:
fix typo

reschke committed:
encoding -> characterEncoding

mattvryan committed:
Merge pull request #2 from reschke/patch-1

Patch 1

Commit:
Change disposition type to always return a value.

Default value is "inline" as was documented before.  This change
makes sure that a value is always set, that inline is the default,
and that the getter is declared as always returning a value.
